app/src/main/java/com/hedvig/app/feature/embark/util/VariableExt.kt

```diff
@@ -42,7 +42,14 @@ fun GraphQLVariablesFragment.AsEmbarkAPIGraphQLSingleVariable.createSingleVariab
 ): Pair<String, Any>? {
     val value = when (as_) {
         EmbarkAPIGraphQLSingleVariableCasting.STRING -> storeValue
-        EmbarkAPIGraphQLSingleVariableCasting.INT -> storeValue.toInt()
+        EmbarkAPIGraphQLSingleVariableCasting.INT -> {
+            try {
+                storeValue.toInt()
+            } catch (exception: NumberFormatException) {
+                // The stored value can in some cases be floats, eg. for computed store values
+                storeValue.toFloat()
```

Comment:
We're requested to cast the variable to `Int`, is this really correct? I think we should probably try to ensure the correct data types on the other border of the data - as the data enters the system - not as the data leaves the system 🙂

Reply:
Yeah agree, but I think that means that we can't return floats from the expression evaluator. Please see my comment above!

Reply:
I see. Still, this would have to be 
```suggestion
                storeValue.toFloat().toInt()
```

right? If not, we would get an upstream API error, providing the API with a `Float` when an `Int` is required.